pkg/common/resource.go

```diff
@@ -99,12 +103,38 @@ func getResource(resourceList v1.ResourceList) *si.Resource {
 	return resources.Build()
 }
 
-func CreateUpdateRequestForTask(appID, taskID string, resource *si.Resource) si.UpdateRequest {
+func createTagsForTask(pod *v1.Pod, taskID string) map[string]string {
+	tags := map[string]string{
+		"namespace": pod.Namespace,
+		"podName": pod.Name,
+	}
+
+	// add Pod labels to Task tags
+	labels, err := json.Marshal(pod.Labels)
```

Comment:
is `labels` a string?
I would prefer to continue to get key/value pairs and store them in the tags map.

Reply:
Labels are key/value pairs with a limited char set content (alphanumerical, dash, underscore and dot)
key is string with a prefix part between 0 and 253 char followed by a slash and a name part which can be 63 char long.
value is a string of 63 char max

Adding them as a marshalled json means that we need to unmarshall on the other side. Keeping them as plain key value pairs is simpler but I do think we need to screen what we pull in.

Reply:
Remove bulk add of all labels